server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPVersion.java

```diff
@@ -0,0 +1,52 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Optional;
+
+public enum JMAPVersion {
+    DRAFT("draft"),
+    RFC8621("rfc-8621");
+
+    public static JMAPVersion of(Optional<String> version) {
+        if (version.isPresent()) {
+            String versionValue = version.get();
```

Comment:
Please rely on functional transformations. This is a flatMap.

Reply:
A flatmap here really?

Reply:
```
public static JMAPVersion of(Optional<String> maybeVersion) {
    return maybeVersion.flapMap(version -> Arrays.stream(values())
                .filter(JMAPVersion::hasValue)
                .findFirst())
     .orElse(DRAFT);
}
```

Yes!

Reply:
Even better, rejecting unknown accept version headers:

```
public static JMAPVersion of(Optional<String> maybeVersion) {
    return maybeVersion.map(version -> JMAPVersion.of(version)
                .orElseThow(() -> new IllegalArgumentExcerption(version + " is not a supported version")))
     .orElse(DRAFT);
}


public static JMAPVersion of(String version) {
    return Arrays.stream(values())
                .filter(jmapVersion -> jmapVersion.hasValue(version));
}
```